### PR TITLE
GH-37290: [MATLAB] Add `arrow.array.Time32Array` class

### LINKED ISSUE #37290: [MATLAB] Add `arrow.array.Time32Array` class
### Describe the enhancement requested

Now that `arrow.type.Time32Type` class has been added to the MATLAB Interface (#37250), we can add the `arrow.array.Time32Array` class.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

Now that the `arrow.type.Time32Type` class has been added to the MATLAB Interface (#37250), we can add the `arrow.array.Time32Array` class. 

### What changes are included in this PR?

1. Added a new class `arrow.array.Time32Array`. It has the following read-only properties: `Type` and `Length`. It has the following methods: `fromMATLAB()`, `toMATLAB()`, and `duration()`. For reference, `duration` arrays in MATLAB represent lengths of time in fixed-length units. Here's a [link](https://www.mathworks.com/help/matlab/ref/duration.html) to the documentation. 
2. Added a new class called `arrow.type.traits.Time32Traits`.

**Example Usage**

```matlab
>> times = seconds([100 120 NaN 200])

times = 

  1×4 duration array

   100 sec   120 sec   NaN sec   200 sec

>> time32Array = arrow.array.Time32Array.fromMATLAB(times)

time32Array = 

[
  00:01:40,
  00:02:00,
  null,
  00:03:20
]

>> roundtrip = toMATLAB(time32Array)

roundtrip = 

  4×1 duration array

   100 sec
   120 sec
   NaN sec
   200 sec

```

You can also specify the `TimeUnit` to use via a name-value pair. `TimeUnit` can either be `"Second"`  (default) or `"Milisecond"`.

```matlab
>> times = seconds([100 120 NaN 200]);
>> time32Array = arrow.array.Time32Array.fromMATLAB(times, TimeUnit="Millisecond")

time32Array = 

[
  00:01:40.000,
  00:02:00.000,
  null,
  00:03:20.000
]
```

### Are these changes tested?

Yes. Added two new test classes: `tTime32Array.m` and `tTime32Traits.m`.

### Are there any user-facing changes?

Yes, users can now create `arrow.array.Time32Array`s from MATLAB `duration`s via the static `fromMATLAB` method on `arrow.array.Time32Array`.

### Future Directions

1. Add `arrow.array.Time64Array`
2. Add `arrow.type.Date32Type` and `arrow.array.Date32Array`
4. Add `arrow.type.Date64Type` and `arrow.array.Date64Array`

* Closes: #37290

### CODE REVIEW

File: matlab/test/arrow/array/tTime32Array.m [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            times = seconds([100 200 NaN 355 NaN 400]);
+            arrray = testCase.ArrowArrayConstructorFcn(times, TImeUnit=Unit);
+            testCase.verifyEqual(arrray.Valid, [true; true; false; true; false; true]);
+            testCase.verifyEqual(toMATLAB(arrray), times');
+            testCase.verifyEqual(duration(arrray), times');
+        end
+
+        function InferNullsTrueNVPair(testCase, Unit)
+            % Verify arrow.array.Time32Array.fromMATLAB() behaves as
+            % expected by InferNulls=true is provided.
```
Comment (kevingurney) [inline matlab/test/arrow/array/tTime32Array.m:120]: by -> when

File: matlab/test/arrow/array/tTime32Array.m [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            array = testCase.ArrowArrayConstructorFcn(times, InferNulls=true, TimeUnit=Unit);
+            expectedValid = [true; true; false; true; true; false; true];
+            testCase.verifyEqual(array.Valid, expectedValid);
+            testCase.verifyEqual(toMATLAB(array), times');
+            testCase.verifyEqual(duration(array), times');
+        end
+
+        function InferNullsFalseNVPair(testCase, Unit)
+            % Verify arrow.array.Time32Array.fromMATLAB() behaves as
+            % expected by InferNulls=false is provided.
```
Comment (kevingurney) [inline matlab/test/arrow/array/tTime32Array.m:132]: by -> when

File: matlab/test/arrow/array/tTime32Array.m [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            % Verify arrow.array.Time32Array.fromMATLAB() accepts any
+            % empty-shaped duration as input.
+
+            times = duration.empty(0, 0);
+            array = testCase.ArrowArrayConstructorFcn(times);
+            testCase.verifyEqual(array.Length, int64(0));
+            testCase.verifyEqual(array.Valid, logical.empty(0, 1));
+            testCase.verifyEqual(toMATLAB(array), duration.empty(0, 1));
+
+            % test with NDimensional empty array
```
Comment (kevingurney) [inline matlab/test/arrow/array/tTime32Array.m:178]: Just a slight wording change suggestion: "Test with an N-Dimensional empty array"

**Review by kevingurney — CHANGES_REQUESTED** [review]
Looks good overall - thank you!

File: matlab/src/cpp/arrow/matlab/array/proxy/time32_array.cc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+       const std::u16string& u16_timeunit = units_mda[0];
+       MATLAB_ASSIGN_OR_ERROR(const auto time_unit,
+                              arrow::matlab::type::timeUnitFromString(u16_timeunit),
+                              error::UKNOWN_TIME_UNIT_ERROR_ID)
+
+       // create the timestamp_type
+       auto data_type = arrow::time32(time_unit);
+       auto array_length = static_cast<int32_t>(timestamp_mda.getNumberOfElements()); // cast size_t to int64_t
+
+       auto data_buffer = std::make_shared<MatlabBuffer>(timestamp_mda);
```
[the remark this answers is not in the document]
Comment (sgilmore10) [reply matlab/src/cpp/arrow/matlab/array/proxy/time32_array.cc:51]: Good catch!

File: matlab/src/matlab/+arrow/+array/Time32Array.m [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+            import arrow.type.TimeUnit
+            import arrow.array.Time32Array
+            import arrow.internal.validate.temporal.timeUnit
+            
+            arrow.internal.validate.type(data, "duration");
+            arrow.internal.validate.shape(data);
+
+            validElements = arrow.internal.validate.parseValidElements(data, opts);
+            time = Time32Array.convertDurationToNumber(data, opts.TimeUnit);
```
[the remark this answers is not in the document]
Comment (sgilmore10) [reply matlab/src/matlab/+arrow/+array/Time32Array.m:77]: I'll change the name!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**